metadatastore/commands.py

```diff
@@ -214,56 +438,27 @@ def insert_run_stop(run_start, time, uid=None, exit_status='success',
     run_stop : mongoengine.Document
         Inserted mongoengine object
     """
-    if uid is None:
-        uid = str(uuid.uuid4())
     if custom is None:
         custom = {}
-    run_start = _get_mongo_document(run_start, RunStart)
-    run_stop = RunStop(run_start=run_start, reason=reason, time=time,
-                       time_as_datetime=_todatetime(time), uid=uid,
-                       exit_status=exit_status, **custom)
 
-    run_stop.save(validate=True, write_concern={"w": 1})
-    logger.debug("Inserted RunStop with uid %s referencing RunStart "
-                 " with uid %s", run_stop.uid, run_start.uid)
-
-    return uid
+    run_start = runstart_given_uid(run_start)
+    runstart_oid = _RUNSTART_UID_to_OID_MAP[run_start['uid']]
+    rs_ref = DBRef('RunStart', runstart_oid)
 
+    run_stop = RunStop(run_start=rs_ref, reason=reason, time=time,
```

Comment:
while we (you) are tinkering with the mds api, what are your feelings on raising if a second insert_run_stop call gets made for the same run_start?

I (amusingly) put the "insert_run_stop" call nested in the event creation loop for one of the dataportal sample scripts
https://gist.github.com/anonymous/d473d5ab69e3d821f9b4 

...and got an integrity error for 21 run stop records :grinning: